django/db/models/query_utils.py

```diff
@@ -220,9 +220,10 @@ def __get__(self, instance, cls=None):
             # might be able to reuse the already loaded value. Refs #18343.
             val = self._check_parent_chain(instance)
             if val is None:
-                if not instance._is_pk_set() and self.field.generated:
+                if not instance._is_pk_set():
```

Comment:
This problem is not specific to generated fields; if the model doesn't have a primary key assigned there is no way to refresh from the database.